static/src/javascripts/projects/common/utils/fetch.js

```diff
@@ -0,0 +1,90 @@
+define([
+    'reqwest',
+    'Promise'
+], function (
+    reqwest,
+    Promise
+) {
+    /**
+     * Provide a minimal function equivalent to fetch. I don't dare calling it a
+     * polyfill but the signature is the same, albeit simplified.
+     *
+     * fetch (input, init): Promise
+     *
+     * Differences with the standard fetch are
+     *
+     * - fetch
+     * - - input can only be a string, Request is not supported
+     * - - does not support JSONP
+     * - - headers can't be specified
+     * - Request
+     * - - there's no way to send a body
+     * - Response
+     * - - blob is not supported
+     * - - formData is not supported
+     * - - headers are not populated
+     *
+     * If you're still wondering what it actually supports
+     * - CORS
+     * - credentials
+     * - response.text() and response.json()
+     * - response.ok .status .statusText
+     */
+    function fetch (input, init) {
+        return new Promise(function(resolve, reject) {
+            var req = buildRequest(input, init || {});
+            reqwest(req)
```

Comment:
Fair enough, I also think this kind of error tracking should happen at a higher level.